deepchem/models/tensorgraph/layers.py

```diff
@@ -4303,3 +4303,38 @@ def batch_mat_mult(self, A, B):
     result = tf.matmul(A_reshape, B)
     result = tf.reshape(result, tf.stack([A_shape[0], A_shape[1], axis_2]))
     return result
+
+
+class Hingeloss(Layer):
+
+  def __init__(self, in_layers=None, weights, batch_size, labels, reg,
+               **kwargs):
+    """
+      Inputs have dimension D, there are C classes, and we operate on minibatches
+      of N examples.
+      Inputs:
+      Weigths: A numpy array of shape (D, C) containing weights.
+      batch_size: A numpy array of shape (N, D) containing a minibatch of data.
+      labels: A numpy array of shape (N,)
+      reg: (float) regularization strength
+    """
+
+    self.weights = weights
+    self.batch_size = batch_size
+    self.labels = labels
+    self.reg = reg
+    super(Hingeloss, self).__init__(in_layers, **kwargs)
+
+  def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
+    inputs = self.get_input_tensors(in_layers)
```

Comment:
Why are we not calling tensorflow's built in hinge loss.
https://www.tensorflow.org/api_docs/python/tf/losses/hinge_loss

Reply:
I thought we wanted to have our own, I will change that